app/Models/ImageMention.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+use App\Models\Concerns\SortableTrait;
+use Illuminate\Support\Arr;
+
+/**
+ * Class ImageMention
+ * @package App\Models
+ *
+ * @property integer $image_id
+ * @property integer $post_id
+ * @property integer $entity_id
+ * @property Entity $entity
+ * @property Post $post
+ * @property Entity $target
+ *
+ * @method static self|Builder prepareCount()
+ */
+class ImageMention extends Model
+{
+    use SortableTrait;
+
+    public $fillable = [
+        'entity_id',
+        'image_id',
+        'post_id',
+    ];
+
+    protected $sortable = [
+        'name',
+        'type',
+    ];
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function target()
+    {
+        return $this->belongsTo('App\Models\Images', 'image_id', 'id');
+    }
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function entity()
+    {
+        return $this->belongsTo('App\Models\Entity', 'entity_id', 'id');
+    }
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function post()
+    {
+        return $this->belongsTo('App\Models\Post', 'post_id', 'id');
+    }
+    
+    /**
+     * Determine if the mention goes to a post
+     * @return bool
+     */
+    public function isPost(): bool
+    {
+        return !empty($this->post_id);
+    }
+
+    /**
+     * Determine if the mention goes to an entity
+     * @return bool
+     */
+    public function isEntity(): bool
+    {
+        return !empty($this->entity_id);
```

Comment:
As per the requirement of an image mention always having an entity_id, this function will always return true, and thus be useless.